core-war/src/main/java/org/silverpeas/web/pdc/control/PdcSearchSessionController.java

```diff
@@ -2172,4 +2168,44 @@ public PagesContext getXMLContext() {
     }
     return pageContext;
   }
+
+  /**
+   * The aim of this provider is to avoid performing several times same component search.<br/>
+   * Another advantage of this provider is to retrieve component data only when it is necessary.
+   */
+  private static class ComponentListProvider implements Serializable {
+    private static final long serialVersionUID = -2487146568792525234L;
+
+    private ComponentSearchCriteria criteria = new ComponentSearchCriteria();
+    private final transient MemoizedSupplier<List<String>> list = new MemoizedSupplier<>(
+        () -> OrganizationController.get().getSearchableComponentsByCriteria(criteria));
+
+    private ComponentListProvider() {
+      // no other way to initialize
+    }
+
+    /**
+     * Sets the search criteria.
+     * <p>
+     *   If criteria are same as previous, then previous search is kept.<br/>
+     *   Otherwise, a new component search will be performed when calling {@link #get()} method.
+     * </p>
+     * @param criteria a component search criteria.
+     */
+    public void setCriteria(final ComponentSearchCriteria criteria) {
```

Comment:
If criteria has to be non null, it should be annotated with `@Nonnull ` annotation (`javax.annotation.Nonnull`)